bindings/matrix-sdk-crypto-js/src/types.rs

```diff
@@ -84,6 +91,42 @@ impl Signatures {
     pub fn count(&self) -> usize {
         self.inner.signature_count()
     }
+
+    /// Get the json with all signatures
+    #[wasm_bindgen(js_name = "asJSON")]
+    pub fn as_json(&self) -> JsValue {
+        trace!(?self.inner, "The signature");
+        // can't use directly serde_wasm_bindgen as there is an issue with BTreeMap
+        // It's always returning an empty {} if I do:
+        // serde_wasm_bindgen::to_value(&self.inner).unwrap()
+
+        // Keep it like that for now as it's working
+        let map: BTreeMap<String, BTreeMap<String, String>> = self
+            .inner
+            .clone()
+            .into_iter()
+            .map(|(u, sign)| {
+                (
+                    u.as_str().to_owned(),
+                    sign.iter()
+                        .map(|(device, maybe_sign)| {
+                            (
+                                device.as_str().to_owned(),
+                                match maybe_sign {
+                                    Ok(s) => s.to_base64(),
+                                    Err(e) => e.source.to_owned(),
+                                },
+                            )
+                        })
+                        .collect(),
+                )
+            })
+            .collect();
+
+        let raw_string = serde_json::to_string(&map).unwrap();
+
+        JSON::parse(&raw_string).unwrap()
+    }
```

Comment:
Again, I'm not really sure why we need this. What is the JS side supposed to do with it?

Reply:
Answer: we have to assemble the `POST /_matrix/client/v3/room_keys/version` request on the application side, and we need the signature data for that.